plugins/optimization-detective/storage/class-od-rest-api.php

```diff
@@ -0,0 +1,360 @@
+<?php
+/**
+ * REST API integration for the plugin: OD_Rest_API.
+ *
+ * @package optimization-detective
+ * @since 0.1.0
+ */
+
+// @codeCoverageIgnoreStart
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+// @codeCoverageIgnoreEnd
+
+/**
+ * OD_Rest_API class
+ *
+ * @since 0.1.0
+ * @access private
+ */
+class OD_Rest_API {
+
+	/**
+	 * Adds hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public static function add_hooks(): void {
+		add_action( 'rest_api_init', array( __CLASS__, 'od_register_endpoint' ) );
+	}
+
+	/**
+	 * Get the namespace for optimization-detective
+	 *
+	 * @since n.e.x.t
+	 * @return non-empty-string Namespace.
+	 */
+	public static function get_namespace(): string {
+		return OD_REST_API_NAMESPACE; // @phpstan-ignore constant.deprecated (To be replaced with string literal when constant is removed.)
+	}
+
+	/**
+	 * Gets the route for storing a URL Metric.
+	 *
+	 * Note the `:store` art of the endpoint follows Google's guidance in AIP-136 for the use of the POST method in a way
+	 * that does not strictly follow the standard usage. Namely, submitting a POST request to this endpoint will either
+	 * create a new `od_url_metrics` post, or it will update an existing post if one already exists for the provided slug.
+	 *
+	 * @since n.e.x.t
+	 * @link https://google.aip.dev/136
+	 * @return non-empty-string Route.
+	 */
+	public static function get_route(): string {
+		return OD_URL_METRICS_ROUTE; // @phpstan-ignore constant.deprecated (To be replaced with string literal when constant is removed.)
+	}
+
+	/**
+	 * Registers endpoint for storage of URL Metric.
+	 *
+	 * @since 0.1.0
```

Comment:
Probably should add a `@since n.e.x.t Moved from od_register_endpoint() function to class method`. Same for other methods in this class.